android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java

```diff
@@ -91,18 +91,19 @@ public void applyMarkdownFormatting(SpannableStringBuilder ssb) {
     try {
       JSONArray ranges = new JSONArray(output);
       for (int i = 0; i < ranges.length(); i++) {
-        JSONArray range = ranges.getJSONArray(i);
-        String type = range.getString(0);
-        int start = range.getInt(1);
-        int end = start + range.getInt(2);
-        applyRange(ssb, type, start, end);
+        JSONObject range = ranges.getJSONObject(i);
+        String type = range.getString("style");
```

Comment:
Let's stick to either `type` or `style`.
```suggestion
        String style = range.getString("style");
```